### PR TITLE
feat(AWS HTTP API): Add support for AWS IAM authorization

### PR DESCRIPTION
**Pull request by pgrzesik**
Add support for `AWS_IAM` authorization for `httpApi`

Closes: #8210

### CODE REVIEW

File: lib/plugins/aws/package/compile/events/httpApi.js [hunk 1]
```diff
@@ -81,6 +81,13 @@ class HttpApiEvents {
             authorizer: {
               anyOf: [
                 { type: 'string' },
+                {
+                  type: 'object',
+                  properties: {
+                    type: { const: 'aws_iam' },
+                  },
+                  additionalProperties: false,
+                },
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/httpApi.js:90]: Configuring definition like that is slightly problematic as it'll trigger _vague_ mode for validation of other authorizers, as now any validation failure will now be reported simply with _"Unsupported configuration format"_ without providing a hint which exactly property is misconfigured.

It's due to fact that two versions for _object_ variant are provided, and validator won't know errors of which to report, hence it'll report just message that something is wrong.

(you may also confirm on that manually).

I think best way to recover is to use one object notation, and then inline ensure that for AWS IAM case it's just `type` property that provided, and throw meaningful error if any other properties are found

**Review by medikoo — CHANGES_REQUESTED** [review]
@pgrzesik looks great! I have just one doubt towards configuration schema setup, please see my comment

File: lib/plugins/aws/package/compile/events/httpApi.js (hunk 1 above)
Comment (pgrzesik) [reply lib/plugins/aws/package/compile/events/httpApi.js]: I've adjusted the configuration so it should be able to accept all possible versions + added additional validation to ensure that we're working with valid configuration.

**Review by medikoo — APPROVED** [review]
Looks great 👍

**Comment by cgguevara** [comment]
is there an example about how to use a external lambda authorizer in httpApi ?

**Pull request merged**